jonahtabb/fun-kickoff-project
# #2 Client feature 2 signup and login style overhaul

### What this PR does:
- Addresses issue #2 
- Major style overhaul to make authorization components match mock-ups.
- Includes major changes to the following components on the Front End:
  - Login, LoginForm, Signup, SignupForm, AuthHeader
- Refactored some common styles to use overrides in the material UI theme provider (theme.ts)
 - This required the use of theme augmentation and installation of Maertial Ui Lab
- Removed username field (based on mock), and adjusted associated front and and backend
- Improvements to mobile responsiveness include:
 - Hamburger menu on tablet and smaller
 - Some additional breakpoint styling

### How to test this PR:
- Fill out and submit sign up form
- Fill out and submit login form
_Both these actions should provide access to the dashboard_

### Outstanding Issues:
- Failing tests that need to be updated.  Mainly due to text changes in the UI / design that were done to match the mock.
- Buttons on mobile nav menu are not functional
- Site automatically attempts to authorize with the server on every page load with a useEffect in UseAuthContext .  I'm not sure if this is expected behavior, or if there is a way to only ping this route if a cookie / data exists.
- Using Roboto as the default font, although the mock font seems different
- Unnecessary commented code in AuthHeader still needs to be removed.

### Screenshots / Videos:
- Loom Video
 <a href="https://www.loom.com/share/e5c601fa907543429d1ec6f031578763"  rel="external" target="_blank">![Foo](https://cdn.loom.com/sessions/thumbnails/e5c601fa907543429d1ec6f031578763-with-play.gif)</a>

- Screenshots
 ![image](https://user-images.githubusercontent.com/80421943/141264509-47f4cdae-2d21-4650-af83-90e5445a91e8.png)
 ![image](https://user-images.githubusercontent.com/80421943/141264612-fce3685d-b199-4dda-8e01-027c145e3196.png)
 ![image](https://user-images.githubusercontent.com/80421943/141264674-fe6d665b-8ccb-485a-9980-6262dd55c20e.png)
